File: onyx-database/src/com/onyx/persistence/annotations/PostUpdate.java
package com.onyx.persistence.annotations;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * This annotation is used to indicate method that is invoked after updating a ManagedEntity.
 *
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * <pre>
 *     <code>
 *      {@literal @}PostUpdate
 *       public void onPostUpdate()
 *       {
 *           Log("Updated Object with id " + this.id);
 *       }
 *     </code>
 * </pre>
 *
 * @see com.onyx.persistence.ManagedEntity
 *
 */
@Target({METHOD})
@Retention(RUNTIME)
public @interface PostUpdate
{
}

File: onyx-database/src/main/kotlin/com/onyx/persistence/annotations/PostUpdate.kt
package com.onyx.persistence.annotations

/**
 * This annotation is used to indicate method that is invoked after updating a ManagedEntity.
 *
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * @PostUpdate
 * public void onPostUpdate()
 * {
 *      Log("Updated Object with id " + this.id);
 * }
 *
 * @see com.onyx.persistence.ManagedEntity
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class PostUpdate
